packages/dataviews/src/dataviews-layouts/grid/composite-grid.tsx

```diff
@@ -0,0 +1,443 @@
+												{ field.header }
+											</FlexItem>
+										</Tooltip>
+										<FlexItem
+											className="dataviews-view-grid__field-value"
+											style={ { maxHeight: 'none' } }
+										>
+											<field.render
+												item={ item }
+												field={ field }
+											/>
+										</FlexItem>
+									</>
+								</Flex>
+							);
+						} ) }
+					</VStack>
+				) }
+			</VStack>
+		</VStack>
+	);
+}
+
+interface CompositeGridProps< Item > {
+	data: Item[];
+	isInfiniteScroll: boolean;
+	className?: string;
+	isLoading?: boolean;
+	view: ViewGridType;
+	fields: NormalizedField< Item >[];
+	selection: string[];
+	onChangeSelection: SetSelection;
+	onClickItem?: ( item: Item ) => void;
+	isItemClickable: ( item: Item ) => boolean;
+	renderItemLink?: (
+		props: {
+			item: Item;
+		} & ComponentProps< 'a' >
+	) => ReactElement;
+	getItemId: ( item: Item ) => string;
+	actions: Action< Item >[];
+}
+
+export default function CompositeGrid< Item >( {
+	data,
+	isInfiniteScroll,
+	className,
+	isLoading,
+	view,
+	fields,
+	selection,
+	onChangeSelection,
+	onClickItem,
+	isItemClickable,
+	renderItemLink,
+	getItemId,
+	actions,
+}: CompositeGridProps< Item > ) {
+	const { paginationInfo, resizeObserverRef } =
+		useContext( DataViewsContext );
+	const gridColumns = useGridColumns();
+	const hasBulkActions = useSomeItemHasAPossibleBulkAction( actions, data );
+	const titleField = fields.find(
+		( field ) => field.id === view?.titleField
+	);
+	const mediaField = fields.find(
+		( field ) => field.id === view?.mediaField
+	);
+	const descriptionField = fields.find(
+		( field ) => field.id === view?.descriptionField
+	);
+	const otherFields = view.fields ?? [];
+	const { regularFields, badgeFields } = otherFields.reduce(
+		(
+			accumulator: Record< string, NormalizedField< Item >[] >,
+			fieldId
+		) => {
+			const field = fields.find( ( f ) => f.id === fieldId );
+			if ( ! field ) {
+				return accumulator;
+			}
+			// If the field is a badge field, add it to the badgeFields array
+			// otherwise add it to the rest visibleFields array.
+			const key = view.layout?.badgeFields?.includes( fieldId )
+				? 'badgeFields'
+				: 'regularFields';
+			accumulator[ key ].push( field );
+			return accumulator;
+		},
+		{ regularFields: [], badgeFields: [] }
+	);
+
+	/*
+	 * This is the maximum width that an image can achieve in the grid. The reasoning is:
+	 * The biggest min image width available is 430px (see /dataviews-layouts/grid/preview-size-picker.tsx).
+	 * Because the grid is responsive, once there is room for another column, the images shrink to accommodate it.
+	 * So each image will never grow past 2*430px plus a little more to account for the gaps.
+	 */
+	const size = '900px';
+	const totalRows = Math.ceil( data.length / gridColumns );
+
+	return (
+		<Composite
+			role={ isInfiniteScroll ? 'feed' : 'grid' }
+			className={ clsx(
+				'dataviews-view-grid',
+				'dataviews-view-grid__content',
```

Comment:
Is there a reason for adding the new `dataviews-view-grid__content` class?

Reply:
I think there was a reason related to dataviews picker, but I searched again and it doesn't use `dataviews-view-grid` classname. So it's fine to remove.